### PR TITLE
STCOM-758 add nonInteractive prop to TextField

### PR DESCRIPTION
**Pull request by aliaksei-chumakou**
https://issues.folio.org/browse/STCOM-758
Guys, here is the another requirement from the ticket - to display text field on the form in non-interactive mode as KeyValue component.

### CODE REVIEW

**Review by ViktorSoroka07 — APPROVED** [review]

**Review by AndreiShumski2 — APPROVED** [review]

**Review by uladislausamets — APPROVED** [review]

**Comment by JohnC-80** [comment]
Why do this here rather via conditional rendering at the App level? If we do this for TextField, then for consistency's sake it would need to be done for all of our individual form controls....

It might be better to create a wrapping component that with either render the KeyValue *or* the Component/Control that's necessary.
The API would look something like: (Just using EditSwitch as temporary name)

```
<EditSwitch interactive component={ TextField } />
```

**Comment by zburke** [comment]
That's a nice API proposal, @JohnC-80, and it encapsulates all the logic in a single component instead of spreading it across every form component.

**Comment by aliaksei-chumakou** [comment]
it might be a good suggestion, but in fact we need this to be implemented exactly for every form control, because non-interactive view is different. For example, Datepicker - it's not a standard value of datefield. The same for Selection. And so on.

File: lib/TextField/TextField.js [hunk 1]
```diff
@@ -92,6 +93,9 @@ class TextField extends Component {
      * String of preset styles to apply to textfield. possible values: noBorder
      */
     inputStyle: PropTypes.string,
+
+    isNonInteractive: PropTypes.bool,
```
Comment (mariia-aloshyna) [inline lib/TextField/TextField.js:97]: Consider adding a description for this prop for consistency, as it's done above for other props.

**Review by mariia-aloshyna — APPROVED** [review]

**Comment by aliaksei-chumakou** [comment]
@JohnC-80  here is the context of requirement https://ux.folio.org/docs/guidelines/ux-patterns/hiding-vs-disabling-elements-ux-pattern/

**Pull request closed without merging**